### PR TITLE
lora/lorawan: Further refactoring

### PR DESCRIPTION
**Pull request by deadprogram**
This PR is to perform further refactoring after #603 has been merged.

### CODE REVIEW

**Comment by deadprogram** [comment]
Any further feedback on this @iamemilio @mirackara @soypat or anyone else?

File: lora/lorawan/adaptor.go [hunk 1]
```diff
@@ -52,13 +52,13 @@ func SetPublicNetwork(enabled bool) {
...
 
 // ApplyChannelConfig sets current Lora modulation according to current regional settings
-func applyChannelConfig(ch *region.Channel) {
-	ActiveRadio.SetFrequency(ch.Frequency)
-	ActiveRadio.SetBandwidth(ch.Bandwidth)
-	ActiveRadio.SetCodingRate(ch.CodingRate)
-	ActiveRadio.SetSpreadingFactor(ch.SpreadingFactor)
-	ActiveRadio.SetPreambleLength(ch.PreambleLength)
-	ActiveRadio.SetTxPower(ch.TxPowerDBm)
+func applyChannelConfig(ch region.Channel) {
```
Comment (iamemilio) [inline lora/lorawan/adaptor.go:55]: The region.Channel object doesn't need to be a pointer, but why copy it?

**Comment by iamemilio** [comment]
Minor Nit pick, but overall LGTM. Much less code duplication

File: lora/lorawan/adaptor.go (hunk 1 above)
Comment (deadprogram) [reply lora/lorawan/adaptor.go:55]: `region.Channel` is now an `interface` hence the change.

**Comment by deadprogram** [comment]
Now merging, thanks for the work on #603 @iamemilio and @mirackara

**Pull request merged**